live-build/misc/upgrade-scripts/verify-container

```diff
@@ -0,0 +1,274 @@
+		"rpool/ROOT/$CONTAINER" ||
+		die "failed to create verification clone"
+
+	mkdir -p "/etc/systemd/nspawn" ||
+		die "failed to create directory: '/etc/systemd/nspawn'"
+
+	#
+	# We need to change some of the default configuration options
+	# for the container, in order for it to behave as we expected:
+	#
+	# * PrivateUsers: We want the UIDs in the container to match the
+	#   UIDs in the host. This way the container can execute commands
+	#   (e.g. zfs) as the normal root user.
+	#
+	# * PrivateUsersChown: Since we set PrivateUsers to "no", we
+	#   need to also ensure systemd-nspawn doesn't attempt to chown
+	#   the containers root filesystem. Normally, with PrivateUsers
+	#   set to "yes", the container's root filesystem would be
+	#   chowned to match the UID mapping used by the container.
+	#
+	# * Private: We set networking to "private" so that the
+	#   container can use ports without conflicting with ports that
+	#   are already in used by the host.
+	#
+	# * Bind=/dev/zfs: We set this so that zfs/zpool/libzpool/etc.
+	#   is usable from within the container.
+	#
+	cat >"/etc/systemd/nspawn/$CONTAINER.nspawn" <<-EOF ||
+		[Exec]
+		PrivateUsers=no
+		[Network]
+		Private=yes
+		[Files]
+		PrivateUsersChown=no
+		Bind=/dev/zfs
+	EOF
+		die "failed to create container configuration file"
+
+	#
+	# Since UPDATE_DIR is stored on a seperate ZFS filesystem than
+	# the root filesysytem, we need to explicitly make this
+	# directory available inside of the verification container,
+	# which is what we're doing here.
+	#
+	# We need UPDATE_DIR to be available inside of the container so
+	# that we can use the APT repository that it contains to upgrade
+	# the packages inside the container, and verify that process.
+	#
+	cat >>"/etc/systemd/nspawn/$CONTAINER.nspawn" <<-EOF ||
+		Bind=$UPDATE_DIR
+	EOF
+		die "failed to add '$UPDATE_DIR' to container config file"
+
+	#
+	# We also need to make "/domain0" available from within the
+	# container, since that's required by the "upgrade-verify" JAR
+	# that's run later. This allows the Virtualization upgrade
+	# verification to verify the contents of it's database (e.g. run
+	# flyway migration) is compatible with the new version of the
+	# Virtualization application.
+	#
+	if [[ -d "/domain0" ]]; then
+		cat >>"/etc/systemd/nspawn/$CONTAINER.nspawn" <<-EOF ||
+			Bind=/domain0
+		EOF
+			die "failed to add '/domain0' to container config file"
+	fi
+
+	#
+	# Bind mounting the DROPBOX_DIR into the container enables any
+	# verification software running in the container to expose their status
+	# and/or results to the host.
+	#
+	# We always want to enable the results of any such verifcation to be
+	# communicated back to the host; thus, we always want to ensure this
+	# directory exists. Otherwise, if the directory didn't exist and we
+	# still attempted to bind mount it into the container, the container
+	# would fail to start.
+	#
+	if [[ ! -d "$DROPBOX_DIR" ]]; then
```

Comment:
Sounds like this is a temporary workaround. I'm assuming that if the dropbox directory isn't there that'd be due to some bug that it would be good to catch.

Reply:
No, this isn't a woraround.

Currently, `DROPBOX_DIR` is created by the verification JAR. We can't do that, since we not run this JAR in the container, so we need to create it outside of that JAR, which is why I do it here.

See [here](http://reviews.delphix.com/r/42619/) for more context.

Reply:
I see. This looks yucky. I feel like we should create DROPBOX_DIR as part of appliance-build.

Reply:
In the `minimal-common` ansible role? or where are you thinking?

Reply:
I think that would be a reasonable place to do it.